packages/react-reconciler/src/ReactFiberLazyComponent.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+type Thenable<T> = {
+  then(resolve: (T) => mixed, reject: (mixed) => mixed): mixed,
+  _reactStatus?: 0 | 1 | 2,
+  _reactResult: any,
+};
+
+export const Pending = 0;
+export const Resolved = 1;
+export const Rejected = 2;
+
+export function getLazyComponentTypeIfResolved<T>(
+  thenable: Thenable<T>,
+): T | null {
+  return thenable._reactStatus === Resolved ? thenable._reactResult : null;
```

Comment:
It seems like this should throw if it is not resolved. If it's not resolved, we can't reason about whether something will be one thing or another so we don't want to make assumptions that might be persisted.

When we do reason about it, we must throw a promise and terminate regardless. 

In cases where we know it'll always be resolved, we want the return type to be specific so that we don't have to add unnecessary checks.

Reply:
I did it this way because it's called from places that aren't allowed to throw. But I suppose I could wrap those call sites in a try-catch.

Reply:
We shouldn't wrap them in try/catch but maybe it's an indication that something else is problematic in those cases. Which ones are they? The issue now is that I can't be sure if something is expected to need a fallback or not.

Reply:
You're right, I got rid of it